Capstone Project One

MrBrN197 opened the pull request:
### Capstone One

- Set up the eslint, webhint, stylelint.
- Setup HTML, CSS, sass files
- Created about and home page.
- Deployed the project
- Create a desktop version for about and home page.
- Implemented mobile menu modal.
- Implemented the featured section with dynamic HTML.

## Video Presentation Demo Link
https://www.loom.com/share/554b31c2303b413fb0465a7f14fc66fa
- In the video, I demonstrate using the desktop and mobile responsive design of the website
- the links that navigate to different parts of the page.

## Code for More-Button and Mobile Menu Feature Demo Link
https://www.loom.com/share/d7e8290bc86d4bdc9f7255a294301ea5
- the "MORE" but is set to show more featured artists when you click on it. 
- the "MORE" icon is set to a loading SVG when clicked is remove from the page when all featured artists are displayed
- the mobile menu that scrolls from the side using javascript and CSS.

MrBrN197 committed:
Add initial basic files

- add a index page
- setup up sass and basic css

MrBrN197 committed:
Add home page html structure

MrBrN197 committed:
track scss files

MrBrN197 committed:
Add about page

MrBrN197 committed:
Add about page footer

MrBrN197 committed:
Add partners section

MrBrN197 committed:
Add `border-top` on each new section

MrBrN197 committed:
Add main-program section css

MrBrN197 committed:
Add css to featured section

MrBrN197 committed:
Fix: h1 padding

MrBrN197 committed:
Replace fonts

- remove old fonts and use new ones
- create font weight sizes for each fonts

MrBrN197 committed:
Start initial desktop version styling

MrBrN197 committed:
Modify flex properties for desktop

MrBrN197 committed:
Create dekstop section margin

MrBrN197 committed:
Create desktop header

MrBrN197 committed:
Add mobile menu popup modal

MrBrN197 committed:
Create featured section dynamically with javascript

MrBrN197 committed:
Add artists to featured section

MrBrN197 committed:
Update README.md

MrBrN197 committed:
Move `setupFeaturedSection` function to it's own file

MrBrN197 committed:
Modify cocogoose.css with correct path

MrBrN197 committed:
Modify html files to link css with relative path

MrBrN197 committed:
Fix: linter errors

MrBrN197 committed:
Fix: github workflows

MrBrN197 committed:
Remove unused font declerations

MrBrN197 committed:
Replace `window.onload` with `addEventListener`

carlos-ssh commented on README.md:
```diff
@@ -1,3 +1,43 @@
+- Styelint
+- CSS/Sass
+
+## Live Demo
```
Kindly work on these requested changes:
**- Provide feedback about the video presentation in the code review summary.** 

1. The maximum length of the video is 5 minutes. 
2. The video contains a demo of the project, showing the pages in different screen sizes. 
3. Provide a highlight about a piece of the code the student finds interesting or is proud of making.

MrBrN197 replied:
I'll do this thank you.

rloterh replied:
Not fully resolved.

MrBrN197 replied:
I've done the third step now. thank you

carlos-ssh commented on about.html:
```diff
@@ -0,0 +1,161 @@
+<!DOCTYPE html>
+<html lang="en">
+
```
- Your project it's missing the About page.

1. The project contains a Home page and About page.
2. It's possible to navigate between the pages, there's a link from Home page to About page and the opposite too.
3. Both pages contain the same navbar and footer.

MrBrN197 replied:
Hi, thanks for the comment. but I have both the home and about page. the home page Is index.html and the about page is about.html. the header has links to both these pages.  can you clarify some more what you mean by the missing About page?

MrBrN197 committed:
Change link paths to use relative paths

carlos-ssh commented on about.html:
```diff
@@ -0,0 +1,161 @@
+</head>
+
+<body>
+  <div id="mobile-menu">
```
**Mobile Size**
The hamburger menu on the left side: 
1. It's background is transparent or the same color/pattern as the top section background.
2. It's missing functionality.

MrBrN197 replied:
1. My hamburger icon is in fact transparent could you clarify more on what you mean with the first point. 
2. with the second point I'm not sure what you mean by missing functionality

rloterh replied:
This issue is well resolved! ✔️

MrBrN197 committed:
Change href paths to relative paths

carlos-ssh commented on about.html:
```diff
@@ -0,0 +1,161 @@
+    </ul>
+  </div>
+  <header>
+    <nav>
```
**Desktop Size**
- Needs to have these:
1. It is divided in a top dark bar and a bottom white bar.
2. The top short bar section with social media links, logout, and language links aligned to the right.
3. The taller bottom bar with a logo and a list of page links.
4. The top bar contains a logo to the left.
5. The top bar contains a list of page links aligned to the right.
![Screen Shot 2021-09-07 at 18 58 49](https://user-images.githubusercontent.com/22258274/132424694-5f3d6fe3-c1bc-4956-9928-13814abd0551.png)

**Yours:**
![Screen Shot 2021-09-07 at 18 59 28](https://user-images.githubusercontent.com/22258274/132424749-3ca4b21b-b1ec-45a4-8914-5a8c2384e056.png)

rloterh replied:
This issue is well resolved! ✔️

MrBrN197 committed:
Fix: href paths

carlos-ssh commented on about.html:
```diff
@@ -0,0 +1,161 @@
+      </ul>
+    </section>
+  </main>
+  <footer>
```
**Footer**
- It contains a logo aligned to the left and text aligned to the right.
The template:
![Screen Shot 2021-09-07 at 19 01 05](https://user-images.githubusercontent.com/22258274/132424848-830f78f7-248a-470b-be4f-0d84754f8fb2.png)

**Yours:**
![Screen Shot 2021-09-07 at 19 01 49](https://user-images.githubusercontent.com/22258274/132424899-0c7cb619-3afa-4a07-81ac-05b3b3f09b95.png)

#
**Mobile Size**
- It has a white background.

**Desktop Size**
- It has a dark background.

rloterh replied:
This issue is well resolved! ✔️

carlos-ssh commented on about.html:
```diff
@@ -0,0 +1,161 @@
+    </div>
+    <ul>
+      <li><a href="/">Home</a></li>
+      <li><a href="/about.html">About</a></li>
```
**Requirements for your About Page:**

- It contains a top section with a centered column that includes a page title, a text block with the page description, and a line with a contact information or any other prompt under it.
- It contains a middle section with a centered column that includes a section header, a divider, a paragraph of text, and an image or large logo.
- It contains a bottom section with a centered column that includes a section header, a divider, a line of text, and two images with text centered over them.

**_Kindly provide the requested style on your Template._**

carlos-ssh commented on index.html:
```diff
@@ -0,0 +1,162 @@
+    </div>
+    <ul>
+      <li><a href="/about.html">About</a></li>
+      <li><a href="/about.html">Program</a></li>
```
[OPTIONAL]
- Disable unused ` <a> ` tags.

MrBrN197 replied:
I'll do this thank you.

carlos-ssh reviewed (changes requested):
Hi @MrBrN197 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

MrBrN197 committed:
Add Google analytics

MrBrN197 committed:
Add `header` banner

MrBrN197 committed:
Make `footer` background dark when on desktop

MrBrN197 committed:
Remove unused `a` tags

MrBrN197 committed:
Fix: stylelint errors

MrBrN197 committed:
Change text to match theme

MrBrN197 committed:
Make mobile menu transition from the side

MrBrN197 committed:
Add the "MORE" `button` functionality

- create a click event listener for the "MORE" button
- remove the "MORE" button when the event is triggered

MrBrN197 committed:
Add loading icon when clicking the MORE button

MrBrN197 committed:
Add linebreaks in `h1`

MrBrN197 committed:
Add closing mobile menu when clicking outside

MrBrN197 committed:
Change main-program icons

MrBrN197 committed:
Change partner section images

MrBrN197 committed:
Change partner images on about page

rloterh commented:
> ### Capstone One
> * Set up the eslint, webhint, stylelint.
> * Setup html, css, sass files
> * Created about and home page.
> * Deployed the project
> * Create desktop version for about and home page.
> * Implemented mobile menu modal.
> * Implemented the featured section with dynamic HTML.
> 
> ## Video Demo Link
> https://www.loom.com/share/554b31c2303b413fb0465a7f14fc66fa

- [x] Good job for adding video presentation of the project, however kindly endeavor to add in the video description some **salient aspects** of your code

rloterh commented on README.md:
```diff
@@ -1,3 +1,43 @@
+- Styelint
+- CSS/Sass
+
+## Live Demo
```
- [x] Good job for adding video presentation of the project, however kindly endeavor to add in the video description some **salient aspects** of your code

MrBrN197 replied:
Thank you I've added this to the description.

rloterh commented on README.md:
```diff
@@ -1,3 +1,43 @@
+
+## Authors
+
+👤 **Author1**
```
- [x] OPTIONAL
Kindly replace the text `**Author1**` with your full name, so as to make your repo more professional

MrBrN197 replied:
Thank You. I've changed this.

rloterh commented on about.html:
```diff
@@ -0,0 +1,184 @@
+  <script async src="https://www.googletagmanager.com/gtag/js?id=G-B899BJXW3P"></script>
+  <script>
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
```
- [x] OPTIONAL
It is a nice idea to prefer `arrow functions ` over normal functions, so as to keep your _function_ syntax simpler and cleaner.

MrBrN197 replied:
It looks like google analytics requires you to using the standard function syntax overusing arrow function so I don't think I can change this without probably causing problems with google analytics.

rloterh commented on about.html:
```diff
@@ -0,0 +1,184 @@
+<!DOCTYPE html>
+<html lang="en">
+
```
- [x] Nice job for adding the _About_ page. However please provide a link on the _About_ page that will send visitors back to the _Home_ page when clicked. You can change the `about menu link` on your _About_ page to a `home link` within the menu.

MrBrN197 replied:
Thank you I've resolved this.

rloterh reviewed (changes requested):
Hi Orora, 

You did great on this project, and for making the changes requested by the previous reviewer! 👏👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦   

## Good Points 👍  
- No linter errors

## Required Changes ♻️
_Kindly check the comments above, as well as below made under the reviews of the previous code reviewer(s) and stating **not properly resolved**. Kindly take your time to correctly review those issues outlined by the previous code reviewer(s), and please check to see that the changes are working as intended._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider the comments marked **[OPTIONAL]** under this review:

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 3 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

MrBrN197 committed:
Update README.md

MrBrN197 committed:
Remove About link from about.html and use Home instead

Davidosky007 commented:
## Hi @MrBrN197 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```
